tests/utils_tests/test_autoreload.py

```diff
@@ -268,3 +276,22 @@ def test_environment(self):
         os.environ['SPAM'] = 'spam'
         with mock.patch.object(sys, 'argv', ['-c', 'pass']):
             autoreload.restart_with_reloader()
+
+    @unittest.skipUnless(six.PY2 and sys.platform == 'win32', 'This is a Python 2 + Windows-specific issue.')
+    def test_environment_decoding(self):
+        """The system encoding is used for decoding."""
+        os.environ['SPAM'] = 'spam'
+        os.environ['EGGS'] = b'\xc6u vi komprenas?'
+        with mock.patch('locale.getdefaultlocale') as default_locale:
+            default_locale.return_value = ('en_US', 'cp1252')
+            with mock.patch.object(sys, 'argv', ['-c', 'pass']):
+                autoreload.restart_with_reloader()
+
+            default_locale.return_value = ('eo', 'latin3')
```

Comment:
Out of curiosity, what's the reason for the two different cases `('en_US', 'cp1252')` and `('eo', 'latin3')`?

Reply:
Thank you.  CP1252 is compatible with Latin-3's C circumflex.  One test for correct mapping, one for compatible, and one for incompatible.

Reply:
Okay, I thought it would be useful to add a comment for each section. Please look at the suggestions, and forgive my ignorance, but is`\xc6` `Æ` ([Latin AE](http://www.codetable.net/hex/c6)) rather than C circumflex?

Reply:
I should have said "non-fatal" instead of "compatible", it's not going to raise an exception but it would result in garbled text.  I updated the comments to reflect that.